debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeTablePointer.java

```diff
@@ -102,34 +101,26 @@ protected boolean isNewTransaction() throws SQLException {
      * a subset of columns
      */
     private ResultSetMapper<Object[]> createResultSetMapper(Table table) throws SQLException {
-        final List<String> sourceTableColumns = table.columns()
-                .stream()
-                .map(Column::name)
-                .collect(Collectors.toList());
+        Map<String, Column> sourceTableColumns = new HashMap<>(table.columns().size());
```

Comment:
This will make the map not big enough due to the default load factor, triggering a re-size if all columns are added. I wouldn't bother and just call new `HashMap<>()`.